### PR TITLE
[Refactor] Common code extraction

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
Clean up of all the common code into the correct places and removal of public LocalizationOptions from the common module

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/State/AppLifeCycleState.swift [hunk 1]
```diff
@@ -19,3 +20,14 @@ struct LifeCycleState {
         self.currentStatus = currentStatus
     }
 }
+
```
Comment (palatter) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/State/AppLifeCycleState.swift:23]: Are we ok with having this swiftUI import in the redux state?

Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/State/AppLifeCycleState.swift:23]: In order to extend EnvironmentKey, we have to.

Comment (waynemo) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/State/AppLifeCycleState.swift:23]: Are we using the extended EnvironmentKey in the redux state? Could we have a separate mapping instead and keep the redux state agnostic of the UI?

**Review by waynemo — DISMISSED** [review]

**Review by Leah-Xia-Microsoft — DISMISSED** [review]

**Review by palatter — DISMISSED** [review]
Looks good.

**Review by vhuseinova-msft — APPROVED** [review]

**Review by jpeng-ms — APPROVED** [review]

**Review by jimchou-dev — APPROVED** [review]
LGTM

**Pull request merged**